installer/scripts/service_control

```diff
@@ -302,167 +590,108 @@ disable_omsagent_service()
+
+	Note: Bourne Shell / Dash / POSIX apparently does not see arguments in a
+	source evocation.  So to get the WORKSPACE_ID to a sourced copy, unless it
+	is in Bourne shell or another shell that sees $# and $1, $2,..., you need to
+	define the variable beforehand, which this draft sees under the name:
+		WORKSPACE_ID 
+EOUSAGE
 }
 
-disable_all_omsagent_services()
-{
-    for ws_id in `ls -1 $ETC_DIR | grep -E $WORKSPACE_REGEX`
-    do
-       setup_variables ${ws_id}
-       disable_omsagent_service
-    done
-
-    check_ws_and_invoke disable_omsagent_service
-}
+# ## Main Procedure
 
 case "$1" in
-    functions)
-        setup_variables $2
-        ;;
-
-    is-running)
-        setup_variables $2
-        is_omsagent_running
-        exit $?
-        ;;
-
-    start)
-        if [ -z $2 ]; then
-            start_all_omsagent
-        else
-            setup_variables $2
-            start_omsagent
-        fi
-        ;;
-
-    stop)
-        if [ -z $2 ]; then
-            stop_all_omsagent
-        else
-            setup_variables $2
-            stop_omsagent
-        fi
-        ;;
-
-    restart)
-        if [ -z $2 ]; then
-            restart_all_omsagent
-        else
-            setup_variables $2
-            restart_omsagent
-        fi
-        ;;
-
-    reload)
-        # TODO: Due to a bug in OMS right now, we can't reload via a signal
-        if [ -z $2 ]; then
-            restart_all_omsagent
-        else
-            setup_variables $2
-            restart_omsagent
-        fi
-        ;;
-
-    find-systemd-dir)
-        find_systemd_dir
-        ;;
-
-    enable)
-        if [ -z $2 ]; then
-            enable_all_omsagent_services
-        else
-            setup_variables $2
-            enable_omsagent_service
-        fi
-        ;;
-
-    disable)
-        if [ -z $2 ]; then
-	    disable_all_omsagent_services
-        else
-            setup_variables $2
-	    disable_omsagent_service
-        fi
-        ;;
-
+    ?|-?|-h|help)			printUsage;;
+    disable)				service_template disable $2;;
+    enable)					service_template enable $2;;
+    find-systemd-dir)		find_systemd_dir;;
+    functions)              setup_variables $2;;
+    is-running)				if setup_variables $2; then
+                                is_omsagent_running
+                                buffer_result=$? # necessary to preserve for the exit contingency below:
+                                return $buffer_result 2>/dev/null || exit $buffer_result
+                            fi;;
+    #TODO: Due to a bug in OMS right now, we can't reload via a signal (Stated Before 2017/08)
+    reload)					service_template restart $2;;
+    restart)				service_template restart $2;;
+    start)					service_template start $2;;
+    stop)					service_template stop $2;;
     *)
         echo "Unknown parameter : $1" 1>&2
-        exit 1
+        printUsage
+        notification_exit $NO_OPTION_PROVIDED
```

Comment:
This should be "invalid operation provided" because a parameter $1 was provided.